packages/dd-trace/test/appsec/iast/analyzers/header-injection.express.plugin.spec.js

```diff
@@ -270,6 +270,128 @@ describe('Header injection vulnerability', () => {
+          fn: (req, res) => {
+            setHeaderFunction('pragma', req.headers['cache-control'], res)
+          },
+          vulnerability: 'HEADER_INJECTION',
+          makeRequest: (done, config) => {
+            return axios.get(`http://localhost:${config.port}/`, {
+              headers: {
+                'Cache-Control': 'cachecontrolvalue'
+              }
+            }).catch(done)
+          }
+        })
+
+        ;['transfer-encoding', 'content-encoding'].forEach((headerName) => {
+          testThatRequestHasVulnerability({
+            testDescription: 'should have HEADER_INJECTION vulnerability when ' +
+              `the header is "${headerName}" and the origin is not a header`,
+            fn: (req, res) => {
+              setHeaderFunction(headerName, req.body.test, res)
+            },
+            vulnerability: 'HEADER_INJECTION',
+            makeRequest: (done, config) => {
+              return axios.post(`http://localhost:${config.port}/`, {
+                test: 'key=value'
+              }, {
+                headers: {
+                  testheader: 'headerValue'
+                }
+              }).catch(done)
+            }
+          })
+
+          testThatRequestHasVulnerability({
+            testDescription: 'should have HEADER_INJECTION vulnerability when ' +
+              `the header is "${headerName}" and the origin is not the accept-encoding header`,
+            fn: (req, res) => {
+              setHeaderFunction(headerName, req.headers.testheader, res)
+            },
+            vulnerability: 'HEADER_INJECTION',
+            makeRequest: (done, config) => {
+              return axios.post(`http://localhost:${config.port}/`, {
+                test: 'key=value'
+              }, {
+                headers: {
+                  testheader: 'headerValue'
+                }
+              }).catch(done)
+            }
+          })
+
+          testThatRequestHasNoVulnerability({
+            testDescription: 'should not have HEADER_INJECTION vulnerability ' +
+              `when the header is "${headerName}" and the origin is a accept-encoding header`,
+            fn: (req, res) => {
+              setHeaderFunction(headerName, req.headers['accept-encoding'], res)
+            },
+            vulnerability: 'HEADER_INJECTION',
+            makeRequest: (done, config) => {
+              return axios.get(`http://localhost:${config.port}/`, {
+                headers: {
+                  'Accept-encoding': 'gzip, deflate'
+                }
+              }).catch(done)
+            }
+          })
+
+          testThatRequestHasNoVulnerability({
+            testDescription: 'should not have HEADER_INJECTION vulnerability ' +
+              `when the header is "${headerName}" and the origin is a substring of accept-encoding header`,
+            fn: (req, res) => {
+              setHeaderFunction(headerName, req.headers['accept-encoding'].split(',').pop(), res)
```

Comment:
I think that you are not testing what you want to test here.
In these type of tests, the rewriter is not modifying the code that is going to be executed, so in `split(',')` you will loose the tainted string.
Maybe, if you export this controller function to other file, move that file to tmp dir and require the new file in the test, it will work as expected.

We are doing it in some tests they need a rewriter, for example: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/test/appsec/iast/analyzers/hardcoded-password-analyzer.spec.js#L108

Reply:
Yes, you are right.

Reply:
Fixed